betting-game-vue/tests/unit/store/matchday.store.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { MatchdayStore } from '@/store/modules/matchday.store'
+import Matchday from '@/models/Matchday'
+import http from '@/utils/Http'
+
+jest.mock('@/utils/Http')
+
+const sampleMatchday: Matchday = {
+  id: 123,
+  name: '1. Spieltag',
+  firstMatchStartDateTime: '2020-10-02T20:30+02:00',
+  lastMatchStartDateTime: '2020-10-04T18:00+02:00',
+  matches: []
+}
+
+describe('Matchday.store', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('Mutations', () => {
+    it('should call FETCH_DATA_PENDING and change to isLoading=true and hasError=false.', () => {
+      const store = new MatchdayStore()
+
+      store.FETCH_DATA_PENDING()
+
+      expect(store.data).toBeUndefined()
+      expect(store.isLoading).toBeTruthy()
+      expect(store.hasError).toBeFalsy()
+      expect(store.errorMsg).toBeUndefined()
+    })
+
+    it('should call FETCH_DATA_SUCCESS and change to isLoading=false and hasError=false and sampleMatchday.', () => {
+      const store = new MatchdayStore()
+
+      store.FETCH_DATA_SUCCESS(sampleMatchday)
+
+      expect(store.data).toEqual(sampleMatchday)
+      expect(store.isLoading).toBeFalsy()
+      expect(store.hasError).toBeFalsy()
+      expect(store.errorMsg).toBeUndefined()
+    })
+
+    it('should call FETCH_DATA_ERROR and change to isLoading=false and hasError=true and error message.', () => {
+      const store = new MatchdayStore()
+
+      store.FETCH_DATA_ERROR('error')
+
+      expect(store.data).toBeUndefined()
+      expect(store.isLoading).toBeFalsy()
+      expect(store.hasError).toBeTruthy()
+      expect(store.errorMsg).toEqual('error')
+    })
+  })
+
+  describe('Actions', () => {
+    describe('getCurrentMatchDay', () => {
+      it('should get the current matchday successfully and provide the corresponsing data', async () => {
+        const store = new MatchdayStore()
+        ;(http.get as jest.Mock).mockReturnValue({
+          data: sampleMatchday
+        })
+
+        await store.getCurrentMatchday()
+
+        expect(store.data).toEqual(sampleMatchday)
+        expect(store.isLoading).toBeFalsy()
+        expect(store.hasError).toBeFalsy()
+        expect(store.errorMsg).toBeUndefined()
+        expect((http.get as jest.Mock).mock.calls[0][0]).toEqual('/matchdays/current')
+      })
+      it('should receive an error while getting the current matchday and provide the error message', async () => {
+        const store = new MatchdayStore()
+        ;(http.get as jest.Mock).mockRejectedValue('error')
```

Comment:
Typically an error thrown by axios is not a string. Instead it has type "AxiosError". Therefore I think it might be good to add an error handling to the http calls. Maybe we should discuss the error format with the backend team as well.

Reply:
You are right, I will change it 👍

Reply:
I changed it to reject an Error() instead of a string and created a separate ticket for the error handling. I think the error handling is necessary in all applications and therefore some more discussion is necessary. But for now the matchday api can be used in the application. I hope it's fine for you.